pyproject.toml

```diff
@@ -34,21 +34,19 @@ types-Deprecated = "^1.2.9"
 pycryptodome = "^3.16.0"
 
 [tool.poetry.dev-dependencies]
-flake8 = "^4.0.1"
-black = "23.3.0"
+flake8 = "^7.0.0"
+black = "24.8.0"
 flake8-black = "^0.3.6"
 flake8-docstrings = "^1.7.0"
 mypy = "^1"
 isort = "^5.11.5"
 flake8-isort = "^6.0.0"
-flake8-annotations = "2.7.0"
```

Comment:
Why is this removed?

Reply:
I have updated the rationale for the removal in the description of the PR.
flake8-annotations depends on an older version of flake8. The solution is to upgrade flake8-annotations to a higher version.

But I couldn't find any usages of this dependency. Where is it used?

Reply:
It isn't used directly, it's used by flake8 to check function annotations: https://github.com/sco1/flake8-annotations?tab=readme-ov-file#table-of-warnings

Reply:
thanks. updated in 563897fb